caikit_nlp/modules/text_generation/text_generation.py

```diff
@@ -0,0 +1,261 @@
+            self._model_loaded = True
+
+        self._bos_token = bos_token
+        self._sep_token = sep_token
+        self._eos_token = eos_token
+        self._pad_token = pad_token
+
+    def __del__(self):
+        # nothing to unload if we didn't finish loading
+        if self._model_loaded:
+            self.get_backend().unload_model(self._model_path)
+
+    @classmethod
+    def bootstrap(cls, base_model_path: str, load_backend: BackendBase = None):
+        """Function to bootstrap a pre-trained transformers model and
+        get a caikit text-generation 'model'.
+
+        Args:
+            base_model_path: str
+                Path to transformers model
+                NOTE: Model path needs to contain tokenizer as well
+            load_backend: BackendBase
+                Backend object to be used to run inference with.
+                NOTE: this is required for inferencing. It is
+                made optional just in provide support for model conversion use-case
+        Returns:
+            caikit_nlp.blocks.text_generation.TextGeneration
+                Object of TextGeneration class (model)
+        """
+        # pylint: disable=duplicate-code
+        model_config = AutoConfig.from_pretrained(base_model_path)
+
+        resource_type = None
+        for resource in cls.supported_resources:
+            if model_config.model_type in resource.SUPPORTED_MODEL_TYPES:
+                resource_type = resource
+                break
+
+        if not resource_type:
+            error(
+                "<FPT61784225E>",
+                "{} model type is not supported currently!".format(
+                    model_config.model_type
+                ),
+            )
+        log.debug("Bootstrapping base resource [%s]", base_model_path)
+        base_model = resource_type.bootstrap(
+            base_model_path, tokenizer_name=base_model_path
+        )
+        bos_token = base_model._tokenizer.bos_token
+        sep_token = base_model._tokenizer.sep_token
+        eos_token = base_model._tokenizer.eos_token or None
+        pad_token = base_model._tokenizer.pad_token
+        return cls(
+            base_model_path,
+            base_model,
+            bos_token=bos_token,
+            sep_token=sep_token,
+            eos_token=eos_token,
+            pad_token=pad_token,
+            tgis_backend=load_backend,
+        )
+
+    def save(self, model_path):
+        """Save caikit model
+
+        Args:
+            model_path: str
+                Folder to save text-generation caikit model
+        """
+        saver = ModuleSaver(
+            self,
+            model_path=model_path,
+        )
+        with saver:
+            artifacts_dir = "artifacts"
+            saver.update_config(
+                {
+                    "artifact_path": artifacts_dir,
+                    "bos_token": self._bos_token,
+                    "sep_token": self._sep_token,
+                    "eos_token": self._eos_token,
```

Comment:
Pad token should also be exportable and reloaded I think

Reply:
yep. good point. Will add that here